operator/src/config.rs

```diff
@@ -679,45 +686,37 @@ pub fn create_nifi_properties(
     format_properties(properties)
 }
 
-pub fn create_state_management_xml(spec: &NifiSpec, zk_ref: &str) -> String {
-    let mut state_management = String::new();
-
-    state_management.push_str("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n");
-    state_management.push_str("<stateManagement>\n");
-
-    state_management.push_str("  <local-provider>\n");
-    state_management.push_str("    <id>local-provider</id>\n");
-    state_management.push_str("    <class>org.apache.nifi.controller.state.providers.local.WriteAheadLocalStateProvider</class>\n");
-    state_management.push_str("    <property name=\"Directory\">./state/local</property>\n");
-    state_management.push_str("    <property name=\"Always Sync\">false</property>\n");
-    state_management.push_str("    <property name=\"Partitions\">16</property>\n");
-    state_management.push_str("    <property name=\"Checkpoint Interval\">2 mins</property>\n");
-    state_management.push_str("  </local-provider>\n");
-
-    state_management.push_str("  <cluster-provider>\n");
-    state_management.push_str("    <id>zk-provider</id>\n");
-    state_management.push_str("    <class>org.apache.nifi.controller.state.providers.zookeeper.ZooKeeperStateProvider</class>\n");
-    // TODO: adapt connect string
-    state_management.push_str(&format!(
-        "    <property name=\"Connect String\">{}</property>\n",
-        zk_ref
-    ));
-    state_management.push_str(&format!(
-        "    <property name=\"Root Node\">{}</property>\n",
+pub fn build_state_management_xml(spec: &NifiSpec, zk_ref: &str) -> String {
+    format!(
+        "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
```

Comment:
Have you tried the raw string syntax that I provided? 
Using that you shouldn't have to escape all the things

Reply:
Yes i had some strange backslashes after the properties. I didnt want to invest too much time, it works probably, but i want to switch to a better xml generation anyways:
https://docs.rs/simple-xml-builder/1.0.0/simple_xml_builder/
https://crates.io/crates/serde_xml_rs